lxd/device/disk.go

```diff
@@ -164,13 +164,23 @@ func (d *disk) checkBlockVolSharing(instanceType instancetype.Type, projectName
 			return nil
 		}
 
-		return db.ErrListStop
-	})
-	if err != nil {
-		if err == db.ErrListStop {
-			return fmt.Errorf("Cannot add custom storage block volume to more than one instance if security.shared is false or unset")
+		// Don't count a VM volume's instance if security.protection.start is preventing that instance from starting
+		if volume.Type == cluster.StoragePoolVolumeTypeNameVM && volume.Project == inst.Project && volume.Name == inst.Name {
+			apiInst, err := inst.ToAPI()
+			if err != nil {
+				return err
+			}
+
+			apiInst.ExpandedConfig = instancetype.ExpandInstanceConfig(d.state.GlobalConfig.Dump(), apiInst.Config, inst.Profiles)
+
+			if shared.IsTrue(apiInst.ExpandedConfig["security.protection.start"]) {
+				return nil
+			}
```

Comment:
> I am wondering now if we should also allow custom block volumes to be attached to more than one instance if .`security.protection.start` is set for all but one of them. If so, it would be nice to keep those consistent I think.

This doesn't really make any sense to me as it has no use though.

Reply:
> If `security.protection.start` is set on the instance using the volume as an additional drive instead of the one using it as root, it should also be okay to attach the volume to it. Am I correct?

I dont follow this scenario, can you expand your idea?

Reply:
@tomponline gets all the credit for this one :wink: 

Making this consistent for all block volumes has some appeal to me; however, it would also mean that an instance can't unset `security.protection.start` until it's checked all of the instances in its project for every volume it currently has attached. [`VolumeUsedByInstanceDevices`](https://github.com/canonical/lxd/blob/main/lxd/storage/utils.go#L1096) would have to be called in a loop and it does a bunch of queries.

Reply:
> This doesn't really make any sense to me as it has no use though.

> it would also mean that an instance can't unset security.protection.start until it's checked all of the instances in its project for every volume it currently has attached.

Yeah this isn't my sharpest thought, sorry just throwing ideas around

Reply:
> I dont follow this scenario, can you expand your idea?

Say we have an instance `vm2` using `vm1` volume as a source for an additional(not-root) disk drive and `vm1` doesn't have `security.shared`. Both `vm1` and `vm2` have `security.protection.start` enabled. Should we allow attaching `vm1` to another instance?

Reply:
> [VolumeUsedByInstanceDevices](https://github.com/canonical/lxd/blob/main/lxd/storage/utils.go#L1096) would have to be called in a loop and it does a bunch of queries.

Though this would also require doing this heavy work when unsetting `security.protection.start`